crates/librqbit/src/blocklist.rs

```diff
@@ -0,0 +1,339 @@
+        if response.status() != 200 {
+            return Err(anyhow::anyhow!(
+                "Failed to fetch blocklist: HTTP {}",
+                response.status()
+            ));
+        }
+
+        let content_length = response
+            .content_length()
+            .ok_or_else(|| anyhow::anyhow!("Failed to get content length"))?;
+
+        if content_length < 2 {
+            return Err(anyhow::anyhow!(
+                "Content too short: not enough data to determine compression"
+            ));
+        }
+
+        let reader = StreamReader::new(
+            response
+                .bytes_stream()
+                .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e)),
+        );
+        Self::create_from_stream(reader).await
+    }
+
+    pub async fn load_from_file(path: &str) -> Result<Self> {
+        let file = tokio::fs::File::open(path).await?;
+        let reader = tokio::io::BufReader::new(file);
+        Self::create_from_stream(reader).await
+    }
+
+    async fn create_from_stream<R>(reader: R) -> Result<Self>
+    where
+        R: AsyncRead + Unpin + Send,
+    {
+        let mut peek_bytes = [0u8; 2];
+        let mut reader = tokio::io::BufReader::new(reader);
+
+        // Peek the first bytes by filling buffer
+        let buffer = reader.fill_buf().await?;
+        if buffer.len() >= 2 {
+            peek_bytes.copy_from_slice(&buffer[0..2]);
+        } else {
+            return Err(anyhow::anyhow!(
+                "Content too short: not enough data to determine compression"
+            ));
+        }
+
+        // Check for Gzip magic bytes (1F 8B)
+        let is_gzip = peek_bytes == [0x1F, 0x8B];
+
+        let reader: Pin<Box<dyn AsyncRead + Send>> = if is_gzip {
+            trace!("Detected Gzip file, decompressing...");
+            Box::pin(BufReader::new(GzipDecoder::new(reader)))
+        } else {
+            trace!("Plain text file detected.");
+            Box::pin(reader)
+        };
+
+        let reader = BufReader::new(reader);
+        let mut lines = reader.lines();
+        let mut ipv4_ranges: Vec<std::ops::Range<u32>> = Vec::new();
+        let mut ipv6_ranges: Vec<std::ops::Range<u128>> = Vec::new();
+        while let Some(line) = lines.next_line().await? {
+            // Skip comments and empty lines
+            if line.starts_with('#') || line.trim().is_empty() {
+                continue;
+            }
+
+            // Parse IP ranges in format: "RuleName:StartIp-EndIp"
+            if let Some((start_ip, end_ip)) = parse_ip_range(&line) {
+                match (start_ip, end_ip) {
+                    (IpAddr::V4(start), IpAddr::V4(end)) => {
+                        let start_num = u32::from_be_bytes(start.octets());
+                        let end_num = u32::from_be_bytes(end.octets());
+                        let range = if end_num == u32::MAX {
+                            start_num..end_num // Special case: Use inclusive range when max
```

Comment:
Can you just use inclusive ranges to simplify this?

```
start_num..=end_num
```

Reply:
Why?

Reply:
IntervalTree [Nodes](https://docs.rs/intervaltree/latest/intervaltree/struct.Element.html) uses Range explicitly. 
So the inclusive range will need to be converted before it is constructed (?).

A better structure for the block list would be a [prefix trie](https://docs.rs/prefix-trie/latest/prefix_trie/index.html) but then a conversion to CIDR representation is needed. If the main concern is not fetching additional dependencies then I can make a naive implementation with existing data structures.

Reply:
Ah, makes sense. It's fine to leave the crate then.